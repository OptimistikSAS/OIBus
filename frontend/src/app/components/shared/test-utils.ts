import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
