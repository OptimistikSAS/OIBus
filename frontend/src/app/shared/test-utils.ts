import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
