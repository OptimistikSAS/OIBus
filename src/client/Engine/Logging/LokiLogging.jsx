import React from 'react'
import PropTypes from 'prop-types'
import OIbForm from '../../components/OIbForm/OIbForm.jsx'
import { minValue, optional } from '../../../services/validation.service'

const schema = { name: 'LokiLogging' }
schema.form = {
  level: {
    type: 'OIbSelect',
    md: 3,
    options: ['none', 'silly', 'debug', 'info', 'warning', 'error'],
    defaultValue: 'info',
    help: <div>The level for the sqlite log</div>,
  },
  host: {
    type: 'OIbText',
    label: 'Host',
    md: 4,
    valid: optional(),
    defaultValue: '',
    help: <div>The host name of the loki instance</div>,
  },
  interval: {
    type: 'OIbInteger',
    newRow: false,
    md: 2,
    valid: minValue(10),
    defaultValue: 60,
    help: <div>Interval between batch of logs (in s)</div>,
  },
  username: {
    type: 'OIbText',
    defaultValue: '',
    valid: optional(),
    md: 3,
  },
  password: {
    newRow: false,
    type: 'OIbPassword',
    defaultValue: '',
    valid: optional(),
    md: 3,
  },
}

const LokiLogging = ({ logParameters, onChange }) => (
  <div>
    <h6>Loki</h6>
    <OIbForm onChange={onChange} schema={schema} name="engine.logParameters.lokiLog" values={logParameters} />
  </div>
)

LokiLogging.propTypes = {
  onChange: PropTypes.func.isRequired,
  logParameters: PropTypes.object.isRequired,
}
export default LokiLogging
