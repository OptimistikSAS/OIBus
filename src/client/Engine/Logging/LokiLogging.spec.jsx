/**
 * @jest-environment jsdom
 */
import React from 'react'
import ReactDOM from 'react-dom'
import { act, Simulate } from 'react-dom/test-utils'

import testConfig from '../../../../tests/testConfig'
import LokiLogging from './LokiLogging.jsx'

const onChange = jest.fn()

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('LokiLogging', () => {
  test('check LokiLogging', () => {
    act(() => {
      ReactDOM.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog level to "warning"', () => {
    act(() => {
      ReactDOM.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.level'), {
      target: {
        value: 'warning',
        selectedIndex: 4,
      },
    })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.level', 'error', null, null)
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog address name to "http://test.loki.fr"', () => {
    act(() => {
      ReactDOM.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.lokiAddress'), { target: { value: 'http://test.loki.fr' } })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.lokiAddress', 'http://test.loki.fr', null)
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog interval to 10', () => {
    act(() => {
      ReactDOM.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.interval'), { target: { value: 10 } })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.interval', 10, null)
    expect(container).toMatchSnapshot()
  })
})
