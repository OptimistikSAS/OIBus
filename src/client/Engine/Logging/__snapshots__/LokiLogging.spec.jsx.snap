// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LokiLogging check LokiLogging 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="none"
            >
              none
            </option>
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.host"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.host"
            name="engine.logParameters.lokiLog.host"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.interval"
          >
            Interval
          </label>
          <input
            aria-invalid="true"
            class="oi-form-input is-invalid form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Value should not greater than 10
          </div>
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog host name to "http://test.loki.fr" 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="none"
            >
              none
            </option>
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.host"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.host"
            name="engine.logParameters.lokiLog.host"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.interval"
          >
            Interval
          </label>
          <input
            aria-invalid="true"
            class="oi-form-input is-invalid form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Value should not greater than 10
          </div>
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog interval to 10 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="none"
            >
              none
            </option>
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.host"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.host"
            name="engine.logParameters.lokiLog.host"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.interval"
          >
            Interval
          </label>
          <input
            aria-invalid="true"
            class="oi-form-input is-invalid form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Value should not greater than 10
          </div>
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog level to "warning" 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="none"
            >
              none
            </option>
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.host"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.host"
            name="engine.logParameters.lokiLog.host"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.lokiLog.interval"
          >
            Interval
          </label>
          <input
            aria-invalid="true"
            class="oi-form-input is-invalid form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Value should not greater than 10
          </div>
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
